### PR TITLE
refactor: move types and utils into their own files

### PR DESCRIPTION
**Pull request by Mike-Heneghan**
## What

- Move the types and utility functions in the `analyticsProvider` file into their own files
- Create an `analytics` folder and update filenames accordingly

## Why 

- Should hopefully improve readability
- Should make testing of utility functions easier

### CODE REVIEW

**Comment by Mike-Heneghan** [comment]
[Improve readability / documentation of analyticsProvider.tsx](https://trello.com/c/zVTLbjUn/2882-improve-readability-documentation-of-analyticsprovidertsx)

File: editor.planx.uk/src/pages/FlowEditor/lib/analyticsMutations.ts
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/lib/analyticsMutations.ts:1]: nit: I know this is still a draft, but I'd favour grouping this in a folder over using a prefix on the filename - e.g. `analytics/types.ts`, `analytics/provider.ts`

Comment (Mike-Heneghan) [reply editor.planx.uk/src/pages/FlowEditor/lib/analyticsMutations.ts:1]: This is a great shout @DafyddLlyr, it should be updated here: 1579eefc066f04c103800cdc29b1f025fd64a016

**Review by DafyddLlyr — APPROVED** [review]
Nice tidy up 👍

**Pull request merged**